### PR TITLE
Auction house instructions indexing

### PR DESCRIPTION
**Pull request by imabdulbasit**

### CODE REVIEW

File: crates/indexer/src/geyser/instructions/buy.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    client
+        .db()
+        .run(move |db| {
+            insert_into(buy_instructions::table)
+                .values(&row)
+                .execute(db)
+        })
+        .await
+        .context("failed to insert buy instruction ")?;
```
Comment (kespinola) [inline crates/indexer/src/geyser/instructions/buy.rs:54]: In future move this to a transaction?

File: crates/indexer/src/geyser/instructions/cancel.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
+use borsh::BorshDeserialize;
+use indexer_core::db::{
+    insert_into,
+    models::CancelInstruction,
+    select,
+    tables::{cancel_instructions, listings, offers},
+    update,
+};
+use mpl_auction_house::instruction::Cancel;
```
Comment (kespinola) [inline crates/indexer/src/geyser/instructions/cancel.rs:9]: Nice getting the cancel struct from the program 👍

**Review by kespinola — APPROVED** [review]
Great work.

**Review by ray-kast — CHANGES_REQUESTED** [review]
Are there any tables we need to add the `check_slot_wv` trigger to?

**Review by ray-kast — CHANGES_REQUESTED** [review]
Looks good.  Still gonna advocate for the use of `check_slot_wv` - we can use `-1` as a fallback for any instructions that don't have these fields.

**Review by ray-kast — APPROVED** [review]

**Pull request merged**